src/rgw/rgw_rados.cc

```diff
@@ -12831,9 +12831,9 @@ int RGWRados::cls_bucket_list(RGWBucketInfo& bucket_info, int shard_id, rgw_obj_
     return r;
 
   // Create a list of iterators that are used to iterate each shard
-  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vcurrents(list_results.size());
-  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vends(list_results.size());
-  vector<string> vnames(list_results.size());
+  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vcurrents;
+  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vends;
+  vector<string> vnames;
```

Comment:
it still makes sense to prealloacate the storage for this vector with `vector::.reserve()` so it doesn't have to reallocate as it grows

Reply:
Yes, you are right @cbodley.
"reserve" is added, would you help to review again? Thanks.